api_fetch.js

```diff
@@ -17,48 +19,70 @@ export default class Challenge {
     imgContainer.classList.add("img__container");
     container.append(imgContainer);
 
-    /* Correct image from api, warning! Crazy cat image! :) Used hero image for each card for now
-            /*
-            const image = document.createElement('img');
-            image.src = this.data.image;
-            image.classList.add("img__container");
-            imgContainer.append(image);
-            */
+    const image = document.createElement("img");
+    image.src = this.data.image;
+    image.classList.add("img__container");
+    imgContainer.append(image);
 
     const titleElement = document.createElement("h2");
     titleElement.textContent = this.data.title + " (" + this.data.type + ")";
     container.append(titleElement);
 
+    // Function to convert rating to stars
+    function createStarRating(rating) {
+      const ratingContainer = document.createElement("div");
+      ratingContainer.classList.add("star__container");
+
+      const fullStar =
+        '<img class="rating__star" src="media/full_star.png" alt="Full star">';
+      const halfStar =
+        '<img class="rating__star" src="media/half_star.png" alt="Half star">';
+      const noStar =
+        '<img class="rating__star" src="media/no_star.png" alt="No star">';
+
+      const fullStars = Math.floor(rating);
+      const hasHalfStar = rating % 1 !== 0;
+
+      for (let i = 0; i < fullStars; i++) {
+        ratingContainer.innerHTML += fullStar;
+      }
+
+      if (hasHalfStar) {
+        ratingContainer.innerHTML += halfStar;
+      }
+
+      const remainingStars = hasHalfStar ? 5 - fullStars - 1 : 5 - fullStars;
```

Comment:
I believe this could have been simplified with `const remainingStars = Math.floor(5 - fullStars);`.